t/helper/test-xml-encode.c

```diff
@@ -0,0 +1,80 @@
+#include "test-tool.h"
+
+static const char *utf8_replace_character = "&#xfffd;";
+
+/*
+ * Encodes (possibly incorrect) UTF-8 on <stdin> to <stdout>, to be embedded
+ * in an XML file.
+ */
+int cmd__xml_encode(int argc, const char **argv)
+{
+	unsigned char buf[1024], tmp[3], *tmp2 = NULL;
```

Comment:
Shouldn't tmp be declared with [4] ?  the 0xF8 case needs 4 chars.

Reply:
Oooops!